src/map.rs

```diff
@@ -4378,6 +4396,24 @@ impl<'a, 'b, K, Q: ?Sized, V, S, A: Allocator> VacantEntryRef<'a, 'b, K, Q, V, S
         &mut entry.1
     }
 
+    /// provide explicit key at insert-time instead of relying on there being effectively a from &K to K implementation and not working with cloneable values
+    #[cfg_attr(feature = "inline-more", inline)]
+    pub fn insert_kv(self, key: K, value: V) -> &'map mut V
+    where
+        K: Hash,
+        for<'k> &'k K: PartialEq<&'key Q>,
```

Comment:
I think this should use `Q: Equivalent<K>` instead, just like `entry_ref` did when it was created.

Reply:
ah yes that covers exactly what is needed there. had not seen that trait so far and tbh i have not read a lot of hashbrowns code, i just needed one of those methods to avoid either copying my key more than needed or doing a double lookup.